src/sql/workbench/services/notebook/common/notebookServiceImpl.ts

```diff
@@ -424,8 +424,8 @@ export class NotebookService extends Disposable implements INotebookService {
 	}
 
 	private waitOnProviderAvailability(providerDescriptor: ProviderDescriptor, timeout?: number): Promise<INotebookProvider> {
-		// Wait up to 10 seconds for the provider to be registered
-		timeout = timeout || 10000;
+		// Wait up to 30 seconds for the provider to be registered
+		timeout = timeout || 30000;
```

Comment:
Let's modify this slightly (or we could put this in the `getProviderInstance`):
- First await `extensionService.whenInstalledExtensionsRegistered`. You'll need to make this a private field instead of it only existing in the constructor. This ensures we wait at least until the extension host is loaded (which could be a long time).
- Then your new timeout still seems reasonable just in case (e.g. if someone is debugging extension startup, or in the future case where we do "no restart needed" extension installs and it then has to take a few seconds to register things